components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/RoleDAOImpl.java

```diff
@@ -1940,6 +1948,57 @@ public int getRoleAudienceRefId(String audience, String audienceId) throws Ident
         return id;
     }
 
+    @Override
+    public List<String> getMainRoleUUIDsForSharedRoles(List<String> sharedRoleUUIDs)
```

Comment:
can't we return a map with shared role id as the key and main role id as the value?
It make more sense and less error-prone

Reply:
Addressed and resolved in commit: [42705b9](https://github.com/wso2/carbon-identity-framework/pull/6162/commits/42705b984942a14ec734891444d31c9d33397a6f)